project/prepare-release.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+set -e
+
+# NOTES:
+#     To delete a tag use: git push origin :v1.0.0
+
+if [[ $1 != "patch" && $1 != "minor" && $1 != "major" ]] ; then
+  echo "    Bad input: \"$1\". Expected an update type of \"patch\", \"minor\" or \"major\"";
+  exit 1;
+fi
+
+echo ""
+echo ""
+echo "Publishing Propel $PACKAGE_VERSION"
+echo ""
+echo ""
+echo "Building Library"
+echo ""
+gulp default --throw-error
```

Comment:
The top of the script - `set -e` bails the script as soon as an error code is returned